src/parser/preprocessor.c

```diff
@@ -60,13 +60,14 @@ struct pp_definition_t {
 	token_t         *token_list;
 
 	position_t       pos;
-	bool             is_expanding    : 1;
-	bool             may_recurse     : 1;
-	bool             has_parameters  : 1;
-	bool             is_parameter    : 1;
-	bool             is_variadic     : 1;
-	bool             standard_define : 1;
-	bool             not_specified   : 1;
+    bool             is_expanding           : 1;
+    bool             may_recurse            : 1;
+    bool             has_parameters         : 1;
+    bool             is_parameter           : 1;
+    bool             is_variadic            : 1;
+    bool             standard_define        : 1;
+    bool             not_specified          : 1;
+    bool             is_specially_handled   : 1; /*predefined macro functions that are handled differntly*/
```

Comment:
- Indent with tabs.
- You could just call the new field `is_pragma`. We can always rename later if we find other cases.